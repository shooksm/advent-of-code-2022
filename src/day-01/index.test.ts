import fs from 'fs';
import { groupDataAndSort, puzzleOne, puzzleTwo } from './index';

const exampleData = `1000
2000
3000

4000

5000
6000

7000
8000
9000

10000`;

const data = fs.readFileSync('./src/day-01/input.txt', 'utf8');

describe('Day 1', () => {
  describe('groupDataAndSort', () => {
    it('should group data, convert to number and sort ascending', () => {
      expect(groupDataAndSort(exampleData)).toEqual([
        4000, 6000, 10000, 11000, 24000,
      ]);
    });
  });

  describe('Puzzle 1', () => {
    it('should return 24000 with example input', () => {
      expect(puzzleOne(exampleData)).toBe(24000);
    });

    it('should calculate the correct answer', () => {
      expect(puzzleOne(data)).toMatchSnapshot();
    });
  });

  describe('Puzzle 2', () => {
    it('should return 45000 with example input', () => {
      expect(puzzleTwo(exampleData)).toBe(45000);
    });

    it('should calculate the correct answer', () => {
      expect(puzzleTwo(data)).toMatchSnapshot();
    });
  });
});
