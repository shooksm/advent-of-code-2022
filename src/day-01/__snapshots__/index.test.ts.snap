// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day 1 Puzzle 1 should calculate the correct answer 1`] = `71780`;

exports[`Day 1 Puzzle 2 should calculate the correct answer 1`] = `212489`;
